external-api/src/cron/eStoreFacilityDirectoryCreationJob.cron.ts

```diff
@@ -0,0 +1,127 @@
+import { HttpStatusCode } from 'axios';
+import { EstoreRepo } from '../repositories/estore/estore-repo';
+import { FacilityFolderResponse, EstoreErrorResponse, Estore } from '../interfaces';
+import { cron } from '../helpers/cron';
+import { ENDPOINT, ESTORE_CRON_STATUS } from '../constants';
+import { createFacilityFolder } from '../v1/controllers/estore/eStoreApi';
+import { getNowAsEpoch } from '../helpers/date';
+
+const ACCEPTABLE_STATUSES = [HttpStatusCode.Ok, HttpStatusCode.Created];
+
+/**
+ * Executes the eStore facility directory creation job.
+ *
+ * This job performs the following tasks:
+ * 1. Checks if facility identifiers are provided in the eStore data.
+ * 2. Creates facility directories using the provided facility identifiers as an array and other deal details.
+ * 3. Updates the cron job logs based on the success or failure of the directory creation.
+ * 4. Initiates the document upload process if the facility directory creation is successful.
+ *
+ * @param {Estore} eStoreData - The eStore data containing information about the deal, exporter, buyer, and facilities.
+ *
+ * @returns {Promise<void>} - A promise that resolves when the job is complete.
+ *
+ * @example
+ * const eStoreData = {
+ *   dealIdentifier: '12345',
+ *   siteId: '507f1f77bcf86cd799439012',
+ *   exporterName: 'Exporter Inc.',
+ *   buyerName: 'Buyer Inc.',
+ *   dealId: '507f1f77bcf86cd799439011',
+ *   facilityIdentifiers: ['FAC123', 'FAC456']
+ * };
+ *
+ * eStoreFacilityDirectoryCreationJob(eStoreData)
+ *   .then(() => console.log('Job completed successfully'))
+ *   .catch((error) => console.error('Job failed', error));
+ */
+export const eStoreFacilityDirectoryCreationJob = async (eStoreData: Estore): Promise<void> => {
+  try {
+    // Argument validation
+    if (
+      !eStoreData?.dealId ||
```

Comment:
const for these?

Reply:
Good spot, apologies I missed this have added.